db/migrate/20150930183738_migrate_content_hosts.rb

```diff
@@ -212,6 +212,9 @@ def get_systems_with_facts(systems)
 
     systems_to_remove.each do |system|
       logger.info("Content Host #{system.uuid} doesn't have candlepin information, unregistering.")
+      system_proxy = SmartProxy.find_by_content_host_id(system.id)
+      system_proxy.content_host = nil
+      system_proxy.save
```

Comment:
I think you need to check for nil here, as not every system will have an associated smart proxy.